### PR TITLE
PMM-9374 external victoria metrics

### PR DESCRIPTION
**Pull request by BupycHuk**
PMM-9374

Link to the Feature Build: SUBMODULES-3331

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

**Review by idoqo — APPROVED** [review]

File: managed/services/supervisord/logs.go [hunk 1]
```diff
@@ -59,14 +61,16 @@ type fileContent struct {
...
 	pmmVersion       string
 	pmmUpdateChecker *PMMUpdateChecker
+	vmParams         *models.VictoriaMetricsParams
 }
 
 // NewLogs creates a new Logs service.
 // n is a number of last lines of log to read.
-func NewLogs(pmmVersion string, pmmUpdateChecker *PMMUpdateChecker) *Logs {
+func NewLogs(pmmVersion string, pmmUpdateChecker *PMMUpdateChecker, vmParams *models.VictoriaMetricsParams) *Logs {
 	return &Logs{
```
Comment (rishat-ishbulatov) [inline managed/services/supervisord/logs.go:70]: Can we separate concrete implementation with abstract one? For example vmParams getter? Than if in future we change mechanism that gets us a new vmParams we wont rewrite this place?

File: managed/services/agents/state.go [hunk 2]
```diff
@@ -41,17 +41,19 @@ const (
...
+	r        *Registry
+	vmdb     prometheusService
+	vmParams *models.VictoriaMetricsParams
 }
 
 // NewStateUpdater creates new agent state updater.
-func NewStateUpdater(db *reform.DB, r *Registry, vmdb prometheusService) *StateUpdater {
+func NewStateUpdater(db *reform.DB, r *Registry, vmdb prometheusService, vmParams *models.VictoriaMetricsParams) *StateUpdater {
 	return &StateUpdater{
-		db:   db,
```
Comment (rishat-ishbulatov) [inline managed/services/agents/state.go:52]: Can we separate concrete implementation with abstract one?

**Review by rishat-ishbulatov — CHANGES_REQUESTED** [review]
cool )

File: managed/models/victoriametrics_params.go [hunk 3]
```diff
@@ -79,3 +93,7 @@ func (vmp *VictoriaMetricsParams) loadVMAlertParams() error {
 
 	return nil
 }
+
+func (vmp *VictoriaMetricsParams) ExternalVM() bool {
+	return !strings.HasPrefix(vmp.URL, "http://127.0.0.1")
+}
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply managed/models/victoriametrics_params.go:99]: 1. Passing a separate environment variable.
2. parsing URL and checking for a host value. if it's 127.0.0.1 or localhost

[the remark this answers is not in the document]
Comment (rishat-ishbulatov) [reply managed/models/victoriametrics_params.go:99]: ok

**Review by rishat-ishbulatov — APPROVED** [review]

**Pull request merged**